src/prolog/lib/sockets.pl

```diff
@@ -23,10 +24,14 @@
     ;
        throw(error(type_error(socket_address, Addr), socket_client_open/3))
     ),
-    builtins:parse_stream_options(Options,
+    (   select(tls(TLS), Options, StreamOptions) ->
```

Comment:
I'd prefer to extend the existing list options parser as follows:

```
parse_socket_options_(tls(TLS), tls-TLS) :-
    must_be(boolean, TLS), !.
parse_socket_options_(Option, OptionPair) :-
    builtins:parse_stream_options_(Option, OptionPair).


parse_socket_options(Options, OptionValues, Stub) :-
    DefaultOptions = [alias-[], eof_action-eof_code, reposition-false, tls-false, type-text],
    builtins:parse_options_list(Options, parse_socket_options_, DefaultOptions, OptionValues, Stub).


socket_client_open(Addr, Stream, Options) :-
    (  var(Addr) ->
       throw(error(instantiation_error, socket_client_open/3))
    ;
       true
    ),
    must_be(var, Stream),
    must_be(list, Options),
    (  Addr = Address:Port,
       atom(Address),
       ( atom(Port) ; integer(Port) ) ->
       true
    ;
       throw(error(type_error(socket_address, Addr), socket_client_open/3))
    ),
    parse_socket_options(Options,
                         [Alias, EOFAction, Reposition, TLS, Type],
                         socket_client_open/3),
    '$socket_client_open'(Address, Port, Stream, Alias, EOFAction, Reposition, Type, TLS).

<rest of file as-is>
```

Reply:
Thank you, I did this!